src/openeo_gfmap/utils/catalogue.py

```diff
@@ -1,16 +1,80 @@
 """Functionalities to interract with product catalogues."""
 import requests
 from geojson import GeoJSON
+from shapely import unary_union
 from shapely.geometry import shape
 
-from openeo_gfmap import SpatialContext, TemporalContext
+from openeo_gfmap import (
+    Backend,
+    BackendContext,
+    BoundingBoxExtent,
+    SpatialContext,
+    TemporalContext,
+)
+
+
+def _parse_cdse_products(response: dict):
+    """Parses the geometry of products from the CDSE catalogue."""
+    geoemetries = []
+    products = response["features"]
+
+    for product in products:
+        geoemetries.append(shape(product["geometry"]))
+    return geoemetries
+
+
+def _query_cdse_catalogue(
+    collection: str,
+    spatial_extent: SpatialContext,
+    temporal_extent: TemporalContext,
+    **additional_parameters: dict,
+) -> dict:
+    if isinstance(spatial_extent, GeoJSON):
+        # Transform geojson into shapely geometry and compute bounds
+        bounds = shape(spatial_extent).bounds
+    elif isinstance(spatial_extent, BoundingBoxExtent):
+        bounds = [
+            spatial_extent.west,
+            spatial_extent.south,
+            spatial_extent.east,
+            spatial_extent.north,
+        ]
+    else:
+        raise ValueError(
+            "Provided spatial extent is not a valid GeoJSON or SpatialContext object."
+        )
+
+    minx, miny, maxx, maxy = bounds
+
+    # The date format should be YYYY-MM-DD
+    start_date = f"{temporal_extent.start_date}T00:00:00Z"
+    end_date = f"{temporal_extent.end_date}T00:00:00Z"
+
+    url = (
+        f"https://catalogue.dataspace.copernicus.eu/resto/api/collections/"
+        f"{collection}/search.json?box={minx},{miny},{maxx},{maxy}"
+        f"&sortParam=startDate&maxRecords=100"
+        f"&dataset=ESA-DATASET&startDate={start_date}&completionDate={end_date}"
+    )
+    for key, value in additional_parameters.items():
+        url += f"&{key}={value}"
+
+    response = requests.get(url)
+
+    if response.status_code != 200:
+        raise Exception(
+            f"Cannot check S1 catalogue on EODC: Request to {url} failed with "
```

Comment:
`EODC` ?

Reply:
Typo fixed,
I added a function to choose the orbit state based on what you just described

Approve the request if this is good for you